aggregate/aggregator.go

```diff
@@ -19,31 +19,46 @@ var (
 )
 
 const (
-	maximumRecordSize       = 1024 * 1024 // 1 MB
-	defaultMaxAggRecordSize = 20 * 1024   // 20K
-	initialAggRecordSize    = 0
-	fieldNumberSize         = 1 // All field numbers are below 16, meaning they will only take up 1 byte
+	defaultMaximumRecordSize = 1024 * 1024 // 1 MB
+	defaultMaxAggRecordSize  = 20 * 1024   // 20K
+	initialAggRecordSize     = 0
+	fieldNumberSize          = 1 // All field numbers are below 16, meaning they will only take up 1 byte
 )
 
 // Aggregator kinesis aggregator
 type Aggregator struct {
-	partitionKeys    map[string]uint64
-	records          []*Record
-	aggSize          int // Size of both records, and partitionKeys in bytes
-	maxAggRecordSize int
-	stringGen        *util.RandomStringGenerator
+	partitionKeys     map[string]uint64
+	records           []*Record
+	aggSize           int // Size of both records, and partitionKeys in bytes
+	maximumRecordSize int
+	maxAggRecordSize  int
+	stringGen         *util.RandomStringGenerator
 }
 
-// NewAggregator create a new aggregator
-func NewAggregator(stringGen *util.RandomStringGenerator) *Aggregator {
+// Config is for aggregation customizations.
+type Config struct {
+	MaximumRecordSize *int
+	MaxAggRecordSize  *int
+}
 
-	return &Aggregator{
+// NewAggregator create a new aggregator
+func NewAggregator(stringGen *util.RandomStringGenerator, cfg *Config) *Aggregator {
+	a := &Aggregator{
 		partitionKeys:    make(map[string]uint64, 0),
 		records:          make([]*Record, 0),
 		maxAggRecordSize: defaultMaxAggRecordSize,
 		aggSize:          initialAggRecordSize,
 		stringGen:        stringGen,
 	}
+
+	if cfg.MaximumRecordSize != nil {
+		a.maximumRecordSize = *cfg.MaximumRecordSize
```

Comment:
do we need to default `a.maximumRecordSize` to `defaultMaximumRecordSize` when `cfg.MaximumRecordSize` is nil ?

Reply:
Ah, I missed it. Thanks, fixed.